tests/integrationv2/common.py

```diff
@@ -78,6 +80,13 @@ def compatible_with_curve(self, curve):
 
         return curve.name[:-3] == self.name[:-3]
 
+    def compatible_with_sigalg(self, sigalg):
+        if self.algorithm is 'EC':
+            if '384' in self.name and 'p256' in sigalg.name:
```

Comment:
The check will behave right, but it's kind of strange. Can we just verify that the sigalg and cert have the same curve in their name? Maybe parse out curve in the class initialization methods like we do other fields?